Connectors/Revit/Speckle.Connectors.RevitShared/HostApp/RevitMaterialBaker.cs

```diff
@@ -0,0 +1,62 @@
+using Autodesk.Revit.DB;
+using Microsoft.Extensions.Logging;
+using Speckle.Converters.RevitShared.Helpers;
+using Speckle.Objects.Other;
+using Speckle.Sdk;
+
+namespace Speckle.Connectors.Revit.HostApp;
+
+public class RevitMaterialBaker
+{
+  private readonly IRevitConversionContextStack _contextStack;
+  private readonly ILogger<RevitMaterialBaker> _logger;
+
+  public RevitMaterialBaker(IRevitConversionContextStack contextStack, ILogger<RevitMaterialBaker> logger)
+  {
+    _contextStack = contextStack;
+    _logger = logger;
+  }
+
+  public Dictionary<string, string> ObjectIdAndMaterialIndexMap { get; } = new();
+
+  public void BakeMaterials(List<RenderMaterialProxy> speckleRenderMaterialProxies)
+  {
+    foreach (var proxy in speckleRenderMaterialProxies)
+    {
+      var speckleRenderMaterial = proxy.value;
+
+      try
+      {
+        var diffuse = System.Drawing.Color.FromArgb(speckleRenderMaterial.diffuse);
+        double transparency = 1 - speckleRenderMaterial.opacity;
+        double smoothness = 1 - speckleRenderMaterial.roughness;
+
+        var newMaterialId = Autodesk.Revit.DB.Material.Create(
+          _contextStack.Current.Document,
+          speckleRenderMaterial.name
+        );
+        var revitMaterial = (Autodesk.Revit.DB.Material)_contextStack.Current.Document.GetElement(newMaterialId);
+        revitMaterial.Color = new Color(diffuse.R, diffuse.G, diffuse.B);
+
+        revitMaterial.Transparency = (int)(transparency * 100);
+        revitMaterial.Shininess = (int)(speckleRenderMaterial.metalness * 128);
+        revitMaterial.Smoothness = (int)(smoothness * 128);
+
+        // do i need something like this?
+        if (speckleRenderMaterial["shine"] is double shine)
+        {
+          revitMaterial.Shininess = (int)(shine * 128);
+        }
+
+        foreach (var objectId in proxy.objects)
+        {
+          ObjectIdAndMaterialIndexMap[objectId] = revitMaterial.UniqueId;
+        }
```

Comment:
ObjectIdAndMaterialIndexMap[objectId] = revitMaterial.Id; > keep track of the elementId so we can use that directly in the mesh converter